libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationRoute.kt

```diff
@@ -0,0 +1,768 @@
+         * @return this builder for chaining options together
+         */
+        fun eventListener(eventListener: EventListener): Builder {
+            directionsBuilder.eventListener(eventListener)
+            return this
+        }
+
+        /**
+         * Enables a route to be refreshable
+         *
+         * @param enableRefresh whether or not to enable refresh
+         * @return this builder for chaining options together
+         */
+        fun enableRefresh(enableRefresh: Boolean): Builder {
+            directionsBuilder.enableRefresh(enableRefresh)
+            return this
+        }
+
+        /**
+         * Sets allowed direction of travel when departing intermediate waypoints. If true the route
+         * will continue in the same direction of travel. If false the route may continue in the
+         * opposite direction of travel. API defaults to true for
+         * [DirectionsCriteria.PROFILE_DRIVING] and false for
+         * [DirectionsCriteria.PROFILE_WALKING] and [DirectionsCriteria.PROFILE_CYCLING].
+         *
+         * @param continueStraight boolean true if you want to always continue straight, else false.
+         * @return this builder for chaining options together
+         */
+        fun continueStraight(continueStraight: Boolean): Builder {
+            directionsBuilder.continueStraight(continueStraight)
+            return this
+        }
+
+        /**
+         * Sets a [NavigationWalkingOptions] object which contains options for use with the
+         * walking profile.
+         *
+         * @param navigationWalkingOptions object holding walking options
+         * @return this builder for chaining options together
+         */
+        fun walkingOptions(navigationWalkingOptions: NavigationWalkingOptions): Builder {
+            directionsBuilder.walkingOptions(navigationWalkingOptions.getWalkingOptions())
+            return this
+        }
+
+        /**
+         * Optionally create a [Builder] based on all variables
+         * from given [RouteOptions].
+         *
+         *
+         * Note: [RouteOptions.bearings] are excluded because it's better
+         * to recalculate these at the time of the request, as your location bearing
+         * is constantly changing.
+         *
+         * @param options containing all variables for request
+         * @return this builder for chaining options together
+         * @since 0.9.0
+         */
+        fun routeOptions(options: RouteOptions): Builder {
+
+            if (!TextUtils.isEmpty(options.baseUrl())) {
+                directionsBuilder.baseUrl(options.baseUrl())
+            }
+
+            if (!TextUtils.isEmpty(options.language())) {
+                directionsBuilder.language(java.util.Locale(options.language()))
+            }
+
+            if (options.alternatives() != null) {
+                directionsBuilder.alternatives(options.alternatives())
+            }
+
+            if (!TextUtils.isEmpty(options.profile())) {
```

Comment:
let's use Kotlin's power 😈 
```suggestion
            if (options.profile().isNotEmpty()) {
```
here and below